### PR TITLE
feat: various fixes and features - october

### LINKED ISSUE #132: [secrets manager] need to exclude aws managed secrets
There are times that secrets can be managed by AWS, they need to be automatically filtered.

Upstream Ref: https://github.com/rebuy-de/aws-nuke/issues/1204

### LINKED ISSUE #133: [docs] add example config templates
Adding common templates that can help users get started that protect common resources like SSO or control tower would be useful. 

Ideas:
- filter SSO
- filter Control Tower

Resources:
- https://github.com/mmunem/dcect/blob/master/cmd/codebuild/reset/default-nuke-config-template.yml

### LINKED ISSUE #301: [NeptuneInstance] can return RDS instances
Should likely implement a filter

```
Name=engine,Values=neptune
```

## References
- https://github.com/rebuy-de/aws-nuke/issues/1094
- https://docs.aws.amazon.com/neptune/latest/userguide/api-instances.html#DescribeDBInstances

### LINKED ISSUE #330: [docs] add user examples for configs
https://github.com/rebuy-de/aws-nuke/issues/711#issuecomment-1170365123

### LINKED ISSUE #334: Add support for OpenSearch Ingestion (OSIS) pipelines
I ran aws-nuke against an account with OpenSearch pipelines and OpenSearch domains, and while aws-nuke removed the OpenSearch domains, it didn't remove the pipelines, and couldn't remove anything that those pipelines depended on, like VPC endpoints.  Please add support for OSIS pipelines to aws-nuke.

See https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/osis for OSIS Pipeline support in the Go AWS SDK

### LINKED ISSUE #367: Do not delete KMS keys by alias
It is possible to specify a filter like this:

```yaml
      KMSAlias:
        - alias/cdk-hnb659fds-assets-key
```

But the underlying key that alias points to, still gets deleted.

```
ap-northeast-1 - KMSKey - c38c7e34-**** - [ID: "c38c7e34-****", Manager: "CUSTOMER", State: "Enabled"] - would remove
```

---

screenshot-20241005T202858-JuzqZwir@2x

### LINKED ISSUE #371: Support deletion of EC2Address with Network Border group
Steps to reproduce:
* create an address that is in a NetworkBorderGroup.

When nuking, if the NetworkBorderGroup isn't specified, the EC2Address isn't released.

> If you provide an incorrect network border group, you receive an InvalidAddress.NotFound

To illustrate, this is the python boto3 code that works:

```python
client = boto3.client('ec2')
response = client.describe_addresses()

for address in response['Addresses']:
    # Disassociate address
    if address.get('AssociationId'):
        client.disassociate_address(
            AssociationId=address['AssociationId']
        )
        print("Disassociated Elastic IP: " + address['AllocationId'])

    client.release_address(
        AllocationId=address['AllocationId'],
        NetworkBorderGroup=address.get('NetworkBorderGroup', '')
    )
    print("Released Elastic IP: " + address['AllocationId'])
    changed = True
```

The go struct `ReleaseAddressInput` to pass that parameter:

```go
type ReleaseAddressInput struct {
	_ struct{} `type:"structure"`

	// The allocation ID. This parameter is required.
	AllocationId *string `type:"string"`

	// Checks whether you have the required permissions for the action, without
	// actually making the request, and provides an error response. If you have
	// the required permissions, the error response is DryRunOperation. Otherwise,
	// it is UnauthorizedOperation.
	DryRun *bool `locationName:"dryRun" type:"boolean"`

	// The set of Availability Zones, Local Zones, or Wavelength Zones from which
	// Amazon Web Services advertises IP addresses.
	//
	// If you provide an incorrect network border group, you receive an InvalidAddress.NotFound
	// error.
	NetworkBorderGroup *string `type:"string"`

	// Deprecated.
	PublicIp *string `type:"string"`
}
```

### LINKED ISSUE #374: APIGatewayRestAPI has delete limit of 1 per 30 seconds

### PR DESCRIPTION
**Pull request by ekristen**
## Overview

Improvements to the documentation around EC2Image and S3Object resource types. Additionally implements rate limiting fixes for API Gateway Rest API and API Key, both of which can only delete 1 resource per 30 seconds. 

- Fixes a bug with `NeptuneInstance` returning non-neptune engine RDS instances.
- Adds new `OSPipeline` resource
- Fixes issue where AWS managed secrets weren't being excluded
- Adds first KMS Alias to KMS Key as property `Alias` for filtering purposes

## References

- Resolves #374 
- Resolves #371
- Resolves #301
- Resolves #132
- Resolves #330
- Resolves #133
- Resolves #334 
- Resolves #367

### CODE REVIEW

**Comment by ekristen** [comment]
Builds available at https://github.com/ekristen/aws-nuke/actions/runs/11282949877

File: docs/resources/ec2-image.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# EC2 Image
+
+This will remove all IAM Roles an AWS account.
```
Comment (mdgm88) [inline docs/resources/ec2-image.md:3]: This will remove all "EC2 Images (AMI)" in an AWS account

**Review by mdgm88 — CHANGES_REQUESTED** [review]

File: docs/resources/ec2-image.md (hunk 1 above)
Comment (ekristen) [reply docs/resources/ec2-image.md:3]: Great catch.

File: docs/config-contrib.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    filters:
+      IAMSAMLProvider:
+        - type: "regex"
+          value: "AWSSSO_.*_DO_NOT_DELETE"
+      IAMRole:
+        - type: "glob"
+          value: "AWSReservedSSO_*"
+      IAMRolePolicyAttachment:
+        - type: "glob"
+          value: "AWSReservedSSO_*"
```
Comment (mdgm88) [inline docs/config-contrib.md:26]: This could also have:

IAMRolePolicy:
  - property: "role:RoleName"
    type: glob
    value: "AWSReservedSSO_*"

**Review by mdgm88 — CHANGES_REQUESTED** [review]

**Review by mdgm88 — CHANGES_REQUESTED** [review]

**Pull request merged**